### DISCUSSION TITLE
Proposal: use publint to validate package exports

### DISCUSSION THREAD

**Discussion by yudai-nkt** [honojs/hono#822] [Ideas]
To avoid bugs like #819 in future development, I'd like to propose using [publint](https://www.npmjs.com/package/publint), which checks validity of exports field in package.json.  I believe this is useful when adding a new exports entry and refactoring existing ones because Hono has many of those. You can try publint in the `v3.0.0-rc.7` tag to see what kind of errors it reports:

```console
$ git checkout v3.0.0-rc.7
$ yarn build
$ npx publint
hono lint results:
Errors:
1. pkg.exports["./cloudflare-pages"].types is ./dist/types/adapter/cloudflare-pages/index.d.ts but file does not exist.
2. pkg.exports["./cloudflare-pages"].import is ./dist/adapter/cloudflare-pages/index.js but file does not exist.
3. pkg.exports["./cloudflare-pages"].require is ./dist/cjs/adapter/cloudflare-pages/index.js but file does not exist.
```

If this is accepted, I'm going to submit a PR based on my [publint](https://github.com/yudai-nkt/hono/tree/publint) branch. What do you think @yusukebe?

**Comment by yusukebe** [comment]
Hi @yudai-nkt !

That's great!! I was struggling with a problem that was only known after it was released to the public. I really want to use "publint". Let's go with it!

**Reply by yudai-nkt** [reply]
Done at #824!
